### PR TITLE
cortex-m/hardfault: add secure-fault handler 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

cortex-m/hardfault: add secure-fault handler

Signed-off-by: chao.an <anchao@xiaomi.com>

## Impact

N/A, new feature

### CODE REVIEW

File: arch/arm/src/armv8-m/arm_securefault.c [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#include "nvic.h"
+#include "sau.h"
+#include "arm_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_SECUREFAULT
+# define sfalert(format, ...)  _alert(format, ##__VA_ARGS__)
```
Comment (pkarashchenko) [inline arch/arm/src/armv8-m/arm_securefault.c:44]:
```suggestion
#  define sfalert(format, ...)  _alert(format, ##__VA_ARGS__)
```

File: arch/arm/src/armv8-m/arm_securefault.c [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+#else
+# define sfalert(x...)
```
Comment (pkarashchenko) [inline arch/arm/src/armv8-m/arm_securefault.c:46]:
```suggestion
#  define sfalert(...)
```

File: arch/arm/src/armv8-m/arm_securefault.c (hunk 1 above)
Comment (anchao) [reply arch/arm/src/armv8-m/arm_securefault.c:44]: Done

File: arch/arm/src/armv8-m/arm_securefault.c [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+#include "arm_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_SECUREFAULT
+#  define sfalert(format, ...)  _alert(format, ##__VA_ARGS__)
+#else
+#  define sfalert(x...)
```
Comment (pkarashchenko) [inline arch/arm/src/armv8-m/arm_securefault.c:46]: Why we need `x...`? I think `(...)` should work

Comment (anchao) [reply arch/arm/src/armv8-m/arm_securefault.c]: Similar definitions I copied from other files.

Comment (pkarashchenko) [reply arch/arm/src/armv8-m/arm_securefault.c]: Ok. Because I saw many use case of `(...)` :)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**